src/lib/db/user-store.ts

```diff
@@ -71,6 +75,35 @@ class UserStore implements IUserStore {
         this.logger = getLogger('user-store.ts');
     }
 
+    async passwordPreviouslyUsed(
+        userId: number,
+        password: string,
+    ): Promise<boolean> {
+        const previouslyUsedPasswords = await this.db(
+            PASSWORD_HASH_TABLE,
+        ).where({ user_id: userId });
+        return previouslyUsedPasswords.some((row) =>
+            bcrypt.compareSync(password, row.password_hash),
+        );
+    }
```

Comment:
Some argument to be made about moving the compareSync into the service instead, so the store only concerns itself with the database op. Will look at come Monday

Reply:
Agreed, would like to see this on the service layer if it's not too much trouble.